### PR TITLE
Routing issues

### LINKED ISSUE #1115: [BUG]: FrontEnd routing issues
### Description

The header in the application routes the application incorrectly and needs to be fixed. If it routes the anchor tag incorrectly, we'll be having a nested routes tree  of /page1/page2/...

### What should have happened?
Its end output must be like this
```html
<a class="dark-white-text menu-text" href="/howitworks">How CiviWiki Works</a>
```
instead of 
```html
<a class="dark-white-text menu-text" href="howitworks">How CiviWiki Works</a>
```

### What browser(s) are you seeing the problem on?
All

### PR DESCRIPTION
**Pull request by SimonDamberg**
I have fixed the routing issues in the header.

Closes #1115

### CODE REVIEW

**Review by brylie — CHANGES_REQUESTED** [review]
While we are working on this related code, would you mind changing the route names to use snake case?

E.g. in the related `urls.py` the route name would be `how_it_works`, `issue_thread`, and `support_us`.

https://github.com/CiviWiki/OpenCiviWiki/blob/3c3282cd091246b041adc6c71099a288d4a77693/project/frontend_views/urls.py#L6-L9

That way, our route names will be consistent.

File: project/core/templates/static_nav.html [hunk 1]
```diff
@@ -13,20 +13,20 @@
 
         <!-- Right Nav -->
         <ul class="right hide-on-med-and-down">
-            <li><a class="dark-white-text menu-text" href="howitworks">{% trans "How CiviWiki Works" %}</a></li>
-            <li><a class="dark-white-text menu-text" href="about">{% trans "About Us" %}</a></li>
-            <li><a class="dark-white-text menu-text" href="support_us">{% trans "Support Us" %}</a></li>
-            <li><a href="login" class="waves-effect waves-light btn login-btn">{% trans "Login / Sign Up" %}</a></li>
+            <li><a class="dark-white-text menu-text" href="{% url 'how it works' %}">{% trans "How CiviWiki Works" %}</a></li>
+            <li><a class="dark-white-text menu-text" href="{% url 'about' %}">{% trans "About Us" %}</a></li>
+            <li><a class="dark-white-text menu-text" href="{% url 'support us' %}">{% trans "Support Us" %}</a></li>
```
Comment (brylie) [inline project/core/templates/static_nav.html]: Let's change these to camel case here and update the related `urls.py` file.
```suggestion
            <li><a class="dark-white-text menu-text" href="{% url 'how_it_works' %}">{% trans "How CiviWiki Works" %}</a></li>
            <li><a class="dark-white-text menu-text" href="{% url 'about' %}">{% trans "About Us" %}</a></li>
            <li><a class="dark-white-text menu-text" href="{% url 'support_us' %}">{% trans "Support Us" %}</a></li>
```

https://github.com/CiviWiki/OpenCiviWiki/blob/3c3282cd091246b041adc6c71099a288d4a77693/project/frontend_views/urls.py#L6-L9

File: project/core/templates/static_nav.html [hunk 2]
```diff
@@ -13,20 +13,20 @@
...
-          <li><a href="login">LOGIN / SIGN UP</a></li>
-          <li><a class="dark-white-text menu-text" href="/">main</a></li>
-          <li><a class="dark-white-text menu-text" href="howitworks">{% trans "How CiviWiki Works" %} </a></li>
-          <li><a class="dark-white-text menu-text" href="about">{% trans "About Us" %} </a></li>
-          <li><a class="dark-white-text menu-text" href="support_us">{% trans "Support Us" %}</a></li>
+            <li><a href="{% url 'accounts_login' %}">LOGIN / SIGN UP</a></li>
+            <li><a class="dark-white-text menu-text" href="/">main</a></li>
+            <li><a class="dark-white-text menu-text" href="{% url 'how it works' %}">{% trans "How CiviWiki Works" %} </a></li>
+            <li><a class="dark-white-text menu-text" href="{% url 'about' %}">{% trans "About Us" %} </a></li>
+            <li><a class="dark-white-text menu-text" href="{% url 'support us' %}">{% trans "Support Us" %}</a></li>
```
Comment (brylie) [inline project/core/templates/static_nav.html]: Let's use snake_case here and be sure to update the related `urls.py`
```suggestion
            <li><a href="{% url 'accounts_login' %}">LOGIN / SIGN UP</a></li>
            <li><a class="dark-white-text menu-text" href="/">main</a></li>
            <li><a class="dark-white-text menu-text" href="{% url 'how_it_works' %}">{% trans "How CiviWiki Works" %} </a></li>
            <li><a class="dark-white-text menu-text" href="{% url 'about' %}">{% trans "About Us" %} </a></li>
            <li><a class="dark-white-text menu-text" href="{% url 'support_us' %}">{% trans "Support Us" %}</a></li>
```

https://github.com/CiviWiki/OpenCiviWiki/blob/3c3282cd091246b041adc6c71099a288d4a77693/project/frontend_views/urls.py#L6-L9

**Review by brylie — CHANGES_REQUESTED** [review]

File: project/core/templates/static_nav.html (hunk 2 above)
Comment (SimonDamberg) [reply project/core/templates/static_nav.html]: I agree! I have updated the PR.

**Comment by SimonDamberg** [comment]
@brylie I have updated the PR according to your requested changes 😄

File: project/core/templates/static_nav.html [hunk 3]
```diff
@@ -13,20 +13,20 @@
...
         <!-- Side Nav for mobile view-->
         <a href="#" data-activates="nav-mobile" class="button-collapse"><i class="material-icons">menu</i></a>
         <ul class="side-nav" id="nav-mobile">
-          <li><a href="login">LOGIN / SIGN UP</a></li>
-          <li><a class="dark-white-text menu-text" href="/">main</a></li>
-          <li><a class="dark-white-text menu-text" href="howitworks">{% trans "How CiviWiki Works" %} </a></li>
-          <li><a class="dark-white-text menu-text" href="about">{% trans "About Us" %} </a></li>
-          <li><a class="dark-white-text menu-text" href="support_us">{% trans "Support Us" %}</a></li>
+            <li><a href="{% url 'accounts_login' %}">LOGIN / SIGN UP</a></li>
+            <li><a class="dark-white-text menu-text" href="/">main</a></li>
```
Comment (gorkemarslan) [inline project/core/templates/static_nav.html]: We have base_view for the main page (`"/"`):

https://github.com/CiviWiki/OpenCiviWiki/blob/3c3282cd091246b041adc6c71099a288d4a77693/project/frontend_views/urls.py#L11

Could you use the route name of this view such that:

`<li><a class="dark-white-text menu-text" href="{% url 'base' %}">main</a></li>`

**Review by gorkemarslan — CHANGES_REQUESTED** [review]

File: project/core/templates/static_nav.html (hunk 3 above)
Comment (SimonDamberg) [reply project/core/templates/static_nav.html]: Fixed this 👍🏼

**Review by gorkemarslan — APPROVED** [review]

**Review by brylie — APPROVED** [review]

**Comment by brylie** [comment]
@all-contributors please add @SimonDamberg for code

**Pull request merged**